### DISCUSSION TITLE
Updating query: how to access a cleaner param than queryKey[1]?

### DISCUSSION THREAD

**Discussion by DoneDeal0** [TanStack/query#1521] [Q&A]
Hi, 

I have a basic query, that I may update depending on a state. It works, but useQuery is passing a queryKey array to my fetch function, which is not ideal and clear.

Is it possible to immediately access a param such as name or id in the get() function?

```javascript
export const getPokemon = (id) => {
  const get = ({ queryKey }) =>
    fetch("https://pokeapi.co/api/v2/pokemon/" + queryKey[1])
        .then((res) =>  res.json());
  const { isLoading, error, data: pokemon } = useQuery(["pokemon", id], get);
  return {
    isLoading,
    error,
    pokemon
  };
};

```

EDIT: ok, I've just found out that you can use ()=> get(id) instead of get. Sorry for the inconvenience, you can delete the question.

**Comment by TkDodo** [comment]
Also, v3 does not spread the queryKey anymore :)
